internal/controller/management_controller.go

```diff
@@ -495,6 +501,100 @@ func (r *ManagementReconciler) ensureAccessManagement(ctx context.Context, mgmt
 	return nil
 }
 
+func (r *ManagementReconciler) ensureStateManagementProvider(ctx context.Context, mgmt *kcmv1.Management) error {
+	l := ctrl.LoggerFrom(ctx).WithValues("provider_name", kcmv1.ProviderSveltosName)
+
+	if _, defined := mgmt.Status.Components[kcmv1.ProviderSveltosName]; !defined {
+		return nil
+	}
+
+	l.Info("Ensuring built-in StateManagementProvider for ProjectSveltos is created")
+	currentNamespace := utils.CurrentNamespace()
+	currentKcmName := os.Getenv("KCM_NAME")
+
+	const (
+		appsAPIGroupVersion = "apps/v1"
+		deploymentKind      = "Deployment"
+
+		projectSveltosDeploymentName      = "addon-controller"
+		projectSveltosDeploymentNamespace = "projectsveltos"
+		projectSveltosAPIGroup            = "config.projectsveltos.io"
+		projectSveltosAPIVersion          = "v1beta1"
+		profilesResource                  = "profiles"
+		clustersummariesResource          = "clustersummaries"
+	)
+
+	stateManagementProvider := &kcmv1.StateManagementProvider{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: utils.DefaultStateManagementProvider,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: kcmv1.GroupVersion.String(),
+					Kind:       mgmt.Kind,
+					Name:       mgmt.Name,
+					UID:        mgmt.UID,
+				},
+			},
+		},
+		Spec: kcmv1.StateManagementProviderSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					utils.DefaultStateManagementProviderSelectorKey: utils.DefaultStateManagementProviderSelectorValue,
```

Comment:
Should this be `currentKcmName` instead of utils.DefaultStateManagementProviderSelectorValue? I’m wondering if it would still work when KCM is deployed with a custom name, since in that case the Deployment name wouldn’t be `kcm-controller-manager`.